workbench-core/accounts/src/services/projectService.ts

```diff
@@ -80,4 +71,204 @@ export default class ProjectService {
+    //   groupId: `${projId}#PA`,
+    //   description: `Project Admin group for ${projId}`
+    // });
+    // const createResearcherGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#Researcher`,
+    //   description: `Researcher group for ${projId}`
+    // });
+    // if (!createProjectAdminGroupResponse.created || !createResearcherGroupResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create Project Admin group or Researcher group with dynamic authorization service.'
+    //   );
+    // }
+
+    // Create Permissions for the groups--TODO implement after dynamic AuthZ
+    // const identityPermissions: IdentityPermission[] = this._createIdentityPermissionsForProject(projectId);
+    // const createIdentityPermissionsResponse = this._dynamicAuthorizationService.createIdentityPermissions(
+    //   user,
+    //   identityPermissions
+    // );
+    // if (!createIdentityPermissionsResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create batch identity permissions for project with dyamic authorization service.'
+    //   );
+    // }
+
+    // Create Proj in DDB
+    const newProject: Project = {
+      id: projectId,
+      createdAt: new Date().toISOString(),
+      costCenterId: params.costCenterId,
+      description: params.description,
+      name: params.name,
+      updatedAt: new Date().toISOString(),
+      status: 'AVAILABLE',
+      // Acc Metadata (get from cost center)
+      subnetId: costCenter.subnetId,
+      vpcId: costCenter.vpcId,
+      envMgmtRoleArn: costCenter.envMgmtRoleArn,
+      externalId: costCenter.externalId,
+      encryptionKeyArn: costCenter.encryptionKeyArn,
+      environmentInstanceFiles: costCenter.environmentInstanceFiles,
+      hostingAccountHandlerRoleArn: costCenter.hostingAccountHandlerRoleArn,
+      awsAccountId: costCenter.awsAccountId,
+      accountId: costCenter.accountId
+    };
+    try {
+      await this._aws.helpers.ddb
+        .update(buildDynamoDBPkSk(projectId, resourceTypeToKey.project), {
+          item: { ...newProject, resourceType: 'project' }
+        })
+        .execute();
+    } catch (e) {
+      console.error('Failed to create project', e);
+      throw Boom.internal('Failed to create project');
+    }
+
+    return newProject;
+  }
+
+  private async _isProjectNameInUse(projectName: string): Promise<void> {
+    // query by name
+    const response = await this._aws.helpers.ddb
+      .query({
+        key: { name: 'resourceType', value: 'project' },
+        index: 'getResourceByName',
+        sortKey: 'name',
+        eq: { S: projectName }
+      })
+      .execute();
+
+    // If anything is responded, name is in use so error
+    if (response.Count !== 0) {
+      throw Boom.badRequest('Project Name is in use by a non deleted project. Please use another name.');
```

Comment:
We can provide a more descriptive error message by including the projectName in the error message.
`Project ${projectName} is in use by a non deleted project. Please use another name`